src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs

```diff
@@ -518,23 +518,18 @@ private void SendEvent(DataCollectionEventArgs args)
                 return;
             }
 
-            foreach (var dataCollectorInfo in this.RunDataCollectors.Values)
-            {
-                if (EqtTrace.IsVerboseEnabled)
-                {
-                    EqtTrace.Verbose("DataCollectionManger:SendEvent: Raising event {0} to collector {1}", args.GetType(), dataCollectorInfo.DataCollectorConfig.FriendlyName);
-                }
 
-                try
-                {
-                    this.events.RaiseEvent(args);
-                }
-                catch (Exception ex)
+            //do not send events multiple times
+
+            try
+            {
+                this.events.RaiseEvent(args);
+            }
+            catch (Exception ex)
```

Comment:
Is `catch` required? It seems we are already using `SafeInvoke` to raise the individual events, it captures and logs the exception?